### PR TITLE
Escrow interface refactor

### PR DESCRIPTION
**Pull request by Psirex**

### CODE REVIEW

File: contracts/Escrow.sol [hunk 1]
```diff
@@ -281,7 +301,110 @@ contract Escrow is IEscrow {
...
+    /// - `claimableAmount`: The amount of claimable ETH contained in the unstETH. This value is 0
+    ///     until the unstETH is finalized or claimed.
+    function getLockedUnstETHDetails(uint256[] calldata unstETHIds)
+        external
+        view
+        returns (LockedUnstETHDetails[] memory unstETHDetails)
+    {
+        unstETHDetails = new LockedUnstETHDetails[](unstETHIds.length);
+
+        for (uint256 i = 0; i < unstETHIds.length; ++i) {
```
[the remark this answers is not in the document]
Comment (Psirex) [reply contracts/Escrow.sol:396]: Revert when the `unstETH` with given id is not locked.

File: contracts/Escrow.sol [hunk 2]
```diff
@@ -463,53 +568,13 @@ contract Escrow is IEscrow {
     }
 
     // ---
-    // Getters
+    // Rage Quit Escrow Getters
```
Comment (Psirex) [inline contracts/Escrow.sol:571]: Rage Quit Escrow: Getters

File: contracts/Escrow.sol [hunk 3]
```diff
@@ -463,53 +568,13 @@ contract Escrow is IEscrow {
...
-
-    /// @notice Returns the total count of unstETH NFTs that have not been claimed yet.
-    /// @return unclaimedUnstETHIdsCount The total number of unclaimed unstETH NFTs.
-    function getUnclaimedUnstETHIdsCount() external view returns (uint256) {
-        _escrowState.checkRageQuitEscrow();
-        return _batchesQueue.getTotalUnclaimedUnstETHIdsCount();
+    /// @notice Returns whether the Rage Quit process has been finalized.
+    /// @return A boolean value indicating whether the Rage Quit process has been finalized (`true`) or not (`false`).
+    function isRageQuitFinalized() external view returns (bool) {
+        return _escrowState.isRageQuitEscrow() && _escrowState.isRageQuitExtensionPeriodPassed();
```
Comment (Psirex) [inline contracts/Escrow.sol:577]:
```suggestion
        _escrowState.checkRageQuitEscrow();
        return _escrowState.isRageQuitExtensionPeriodPassed();
```

File: contracts/Escrow.sol [hunk 4]
```diff
@@ -520,46 +585,25 @@ contract Escrow is IEscrow {
         unstETHIds = _batchesQueue.getNextWithdrawalsBatches(limit);
     }
 
-    /// @notice Returns whether all withdrawal batches have been closed.
-    /// @return isWithdrawalsBatchesClosed A boolean value indicating whether all withdrawal batches have been
-    ///     closed (`true`) or not (`false`).
-    function isWithdrawalsBatchesClosed() external view returns (bool) {
+    /// @notice Retrieves details about the current state of the rage quit escrow.
+    /// @return details A `RageQuitEscrowDetails` struct containing the following fields:
+    /// - `isWithdrawalsBatchesClosed`: Indicates whether the withdrawals batches are closed.
```
Comment (Psirex) [inline contracts/Escrow.sol:590]: Move `isWitdrawalsBatchesClosed` and `isRageQuitExtensionPeriodStarted` into standalone getters

**Review by bulbozaur — APPROVED** [review]

**Pull request merged**